diesel/src/sqlite/connection/stmt.rs

```diff
@@ -58,8 +58,15 @@ impl Statement {
         unsafe { ffi::sqlite3_column_count(self.inner_statement.as_ptr()) as usize }
     }
 
-    /// The lifetime of the returned CStr is shorter than self. This function
-    /// should be tied to a lifetime that ends before the next call to `reset`
+    /// The lifetime of the returned CStr is shorter than self.
+    ///
+    /// > The returned string pointer is valid until either the prepared statement
+    /// > is destroyed by sqlite3_finalize() or until the statement is automatically
+    /// > reprepared by the first call to sqlite3_step() for a particular
+    /// > run or until the next call to sqlite3_column_name()
```

Comment:
I don't understand what "for a particular run" means in this context. I'd be able to review more usefully if I understood precisely what this means :)

Reply:
That's the exact wording of the SQLite documentation for this function. I interpret this in such a way that the pointer for a given column name can change if you execute a prepared statement and after the executing call `sqlite3_step()` the first time for this newly executed statement. (`sqlite3_step()` is basically `Iterator::next()` to get a new row from the result set). So we should fine as long as we call `sqlite3_column_name()` after the first call to `sqlite3_step()` for a particular run.

Reply:
Hmm so if I understand correctly "for a particular run" means for us "for an instance of `StatementUse`".

Reply:
Thats correct :+1: